cmd/snap-confine/snap-confine.c

```diff
@@ -358,9 +358,7 @@ int main(int argc, char **argv)
 		sc_error *err SC_CLEANUP(sc_cleanup_error) = NULL;
 		snap_context =
 		    sc_cookie_get_from_snapd(invocation.snap_instance, &err);
-		if (err != NULL) {
-			error("%s\n", sc_error_msg(err));
-		}
+		/* The error is explicitly ignored because the cookies are optional. */
```

Comment:
I think the change is fine, what that means in practice is we will only see an error if the snap actually attempts to run snapctl from its app (and the cookie is missing). In such case snapctl should report "invalid snap cookie requested" if I read the code correctly (the critical bit is in `func (m *HookManager) Context(cookieID string) (*Context, error) {...` and `func (m *HookManager) ephemeralContext(cookieID string) (context *Context, err error) {....` in hookmgr.go, called from `runSnapctl` in api.go. But this is theoretical, because `SyncCookies()` run when overlord starts makes sure all cookies are created if missing after an upgrade from old snapd.

The only remark I've is the comment here, I think it's slightly misleading. Could you please expand it a little bit by mentioning SyncCookies, snapctl error and that cookies are only needed if snapctl is run by an app?